src/components/dropdown/daikin-dropdown.ts

```diff
@@ -0,0 +1,373 @@
+   */
+  @property({ type: String, reflect: true })
+  placeholder = "";
+
+  /**
+   * Whether the dropdown is disabled
+   */
+  @property({ type: Boolean, reflect: true })
+  disabled = false;
+
+  /**
+   * Whether the dropdown is required
+   */
+  @property({ type: Boolean, reflect: true })
+  required = false;
+
+  /**
+   * Whether the dropdown is error
+   */
+  @property({ type: Boolean, reflect: true })
+  error = false;
+
+  /**
+   * Whether or not a drop-down menu is displayed
+   */
+  @property({ type: Boolean, reflect: true })
+  open = false;
+
+  @queryAssignedElements({ selector: "daikin-dropdown-item" })
+  private readonly _items!: readonly DaikinDropdownItem[];
+
+  @state()
+  private _hasSelectedItem = false;
+
+  @state()
+  private _selectedItemLabel = "";
+
+  private _autoUpdateController = new FloatingUIAutoUpdateController(this);
+
+  private _clickOutsideController = new ClickOutsideController(
+    this,
+    this._onClickOutside
+  );
+
+  private _onClickOutside(): void {
+    this.open = false;
+  }
+
+  private _reflectItemsAndLabel(): void {
+    const items = this._items;
+
+    const selectedItem = items.find((item) => item.value === this.value);
+    for (const item of items) {
+      item.selected = item === selectedItem;
+    }
+
+    this._hasSelectedItem = !!selectedItem;
+    this._selectedItemLabel = selectedItem?.textContent ?? "";
+  }
+
+  private _handleClick(): void {
+    this.open = !this.open;
+  }
+
+  private _handleKeyDown(event: KeyboardEvent): void {
+    const operation = (
+      {
+        ArrowDown: "moveDown",
+        ArrowUp: "moveUp",
+        Escape: "close",
+      } as const
+    )[event.key];
+
+    if (!operation) {
+      return;
+    }
+
+    if (operation === "close") {
+      if (this.open) {
+        // Close
+        this.open = false;
+      } else {
+        // Clear selection
+        this.value = undefined;
+      }
+      return;
+    }
+
+    const moveOffset = operation === "moveUp" ? -1 : 1;
+    const items = this._items;
+
+    // Get focused item if any
+    const activeElement = document.activeElement;
+    const focusedItemIndex = activeElement
+      ? items.findIndex((item) => item.contains(activeElement))
+      : -1;
+
+    // If there is no dropdown focused, do nothing.
+    if (focusedItemIndex === -1 && moveOffset === -1) {
+      return;
+    }
+
+    // Focus on the first dropdown that is enabled.
+    for (
+      let index = focusedItemIndex + moveOffset, i = 0;
+      i < items.length;
+      index += moveOffset, i++
+    ) {
+      index =
+        Math.sign(index % items.length) === -1
+          ? items.length - 1
+          : index % items.length;
+      const item = items[index];
+
+      if (!this.open) {
+        this.open = true;
+      }
+
+      if (item.disabled) {
+        continue;
+      }
+
+      this.updateComplete
+        .then(() => item.focus())
+        .catch((error: unknown) => console.error(error));
```

Comment:
nit. We don't have to do `console.error`.

```suggestion
      this.updateComplete
        .then(() => {
          item.focus();
        }, () => {
         // do nothing
        });
```